scanpipe/models.py

```diff
@@ -1729,6 +1729,9 @@ class DiscoveredPackage(
     package_uid = models.CharField(
         max_length=1024,
         blank=True,
+        unique=True,
```

Comment:
We cannot use a simple `unique=True` here as it would prevent to load the same inventory into 2 project for example.
I would suggest that we make the `package_uid` unique in the context of a project.
We can use this constrain: https://docs.djangoproject.com/en/4.0/ref/models/constraints/#django.db.models.UniqueConstraint

Reply:
@JonoYang The migration in https://github.com/nexB/scancode.io/pull/454/commits/ee2453cbe530391eef9e21d5e952350e29b3e779 creates the following DB indexes and will ensure the unicity of `package_uid`s inside a Project:
```
CREATE INDEX "scanpipe_discoveredpackage_package_uid_6b082896" ON "scanpipe_discoveredpackage" ("package_uid");
CREATE INDEX "scanpipe_discoveredpackage_package_uid_6b082896_like" ON "scanpipe_discoveredpackage" ("package_uid" varchar_pattern_ops);
CREATE UNIQUE INDEX "scanpipe_discoveredpackage_unique_package_uid_within_project" ON "scanpipe_discoveredpackage" ("project_id", "package_uid") WHERE NOT ("package_uid" = '');
```